### PR TITLE
[ADD] sudan public holidays

### PR DESCRIPTION
**Pull request by Mazin-Ibrahim**
# Contributing a new country ?

* [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/spatie/holidays/pulls) for the same country?

1. Create a new class in the Countries directory. It should extend the Country class.
2. Add a test for the new country in the tests directory.
3. Run the tests so a snapshot gets created.
4. Verify the result in the newly created snapshot is correct.

### CODE REVIEW

**Comment by Mazin-Ibrahim** [comment]
[ADD] sudan public holidays

**Comment by Nielsvanpach** [comment]
Could you use this implementation? https://github.com/spatie/holidays/pull/138/files

File: src/Countries/Sudan.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+            'Christmas Day' => '12-25'
+        ], $this->variableHolidays($year));
+    }
+
+    /**
+     * @return array<string, CarbonImmutable>
+     */
+    private function variableHolidays(int $year): array
+    {
+        return [];
```
Comment (Nielsvanpach) [inline src/Countries/Sudan.php:29]: This is how other countries solved the Islamic holidays: https://github.com/spatie/holidays/blob/main/src/Countries/Turkey.php#L8

**Comment by Nielsvanpach** [comment]
Thanks for your effort, but I'm closing this PR as the Islamic holidays are missing. 
Feel free to reopen if you decide to continue this PR!

**Pull request closed without merging**